test/features/number_trivia/domain/usecases/get_concrete_number_test.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:clean_code_architecture_tdd/features/number_trivia/domain/entities/number_trivia.dart';
+import 'package:clean_code_architecture_tdd/features/number_trivia/domain/repositories/number_trivia_repository.dart';
+import 'package:clean_code_architecture_tdd/features/number_trivia/domain/usecases/get_concrete_number_trivia.dart';
+import 'package:dartz/dartz.dart';
+import 'package:mockito/mockito.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class MockNumberTriviaRepository extends Mock
+    implements NumberTriviaRepository {}
+
+void main() {
+  GetConcreteNumberTrivia usecase;
+  MockNumberTriviaRepository mockNumberTriviaRepository;
+  setUp(() {
+    mockNumberTriviaRepository = MockNumberTriviaRepository();
+    usecase = GetConcreteNumberTrivia(mockNumberTriviaRepository);
+  });
+
+  final tNumber = 1;
+  final tNumberTrivia = NumberTrivia(text: "test", number: 1);
+  test('should', () async {
+    // arrange
+    mockNumberTriviaRepository = MockNumberTriviaRepository();
+    when(mockNumberTriviaRepository.getConcreteNumberTrivia(2))
```

Comment:
Failing: type 'Null' is not a subtype of type 'Future<Either<Faliure, NumberTrivia>>